transport/src/main/java/io/netty/channel/AbstractChannel.java

```diff
@@ -937,16 +874,95 @@ protected final Throwable annotateConnectException(Throwable cause, SocketAddres
 
             return cause;
         }
+    }
 
-        /**
-         * Prepares to close the {@link Channel}. If this method returns an {@link Executor}, the
-         * caller must call the {@link Executor#execute(Runnable)} method with a task that calls
-         * {@link #doClose(ChannelPromise)} on the returned {@link Executor}. If this method returns {@code null},
-         * {@link #doClose(ChannelPromise)} must be called from the caller thread. (i.e. {@link EventLoop})
-         */
-        protected Executor prepareToClose() {
-            return null;
+    protected final void handleWriteError(Throwable t) {
+        ioTransport.handleWriteError(t);
+    }
+
+    /**
+     * Prepares to close the {@link Channel}. If this method returns an {@link Executor}, the
+     * caller must call the {@link Executor#execute(Runnable)} method with a task that calls
+     * {@link #doClose(ChannelPromise)} on the returned {@link Executor}. If this method returns {@code null},
+     * {@link #doClose(ChannelPromise)} must be called from the caller thread. (i.e. {@link EventLoop})
+     */
+    protected Executor prepareToClose() {
+        return null;
+    }
+
+    /**
+     * Shutdown the output portion of the corresponding {@link Channel}.
+     * For example this will clean up the {@link ChannelOutboundBuffer} and not allow any more writes.
+     */
+    // TODO: Make this a concept of the Unsafe.
```

Comment:
Of the now-hidden Unsafe? Got plans to address this?

Reply:
Yes I will port this https://github.com/netty/netty/commit/d3cf5bea6d7b0c1d78e207bd09c615a8c2da579c to address it. Will do it soon